app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -83,6 +80,16 @@ def complete
     redirect_to @origin
   end
 
+  def valid_origin?(uri)
+    if uri && (uri.host == nil || uri.host == Discourse.current_hostname) # valid & local
+      return false if uri.path.start_with?(Discourse.base_path("/auth/"))
+      return false if uri.path.start_with?(Discourse.base_path("/login"))
```

Comment:
I've fallen into this trap before - the argument to `Discourse.base_path` is "default_value", not "path". On a subfolder site, these calls will just return something like `/forum`, without the `/auth` or `/login` segments